scripts/generate_validation_data_xlinear.py

```diff
@@ -0,0 +1,72 @@
+""" Script to generate validation data from Wellcome Grants """
+import os
+
+import pandas as pd
+
+try:
+    from datascience.warehouse.warehouse import FortyTwo
+    from datascience.grants.cleaning import clean_grants
+except ImportError as e:
+    raise ImportError(f"Error importing {e.name }. "
+                      f"To use this script you need to install Wellcome's datascience internal "
+                      f"utils, for example with `make install-private-requirements`")
+
+
+# Name of the file output of the pipeline
+predictions_file = 'mesh_pipeline_result.csv'
+
+# Name of the file with the predicitons, alongside metadata
+merged_predictions_file = 'merged_mesh_predictions_mesh_xlinear_for_validation.xlsx'
+here = os.path.abspath(os.path.dirname(__file__))
+
+grants = pd.read_csv(os.path.join(here, '../data/raw/grants.csv'), index_col=0)
+grants = clean_grants(grants).fillna('')
+
+## Query FortyTwo to obtain the MeSH descriptors (could query the raw XML too?)
+descriptor_query = """
+SELECT [MeSH Lowest Level Classification], [MeSH Classification Description]
+FROM FortyTwo.dbo.MeSHClassificationHierarchy 
+"""
+
+forty_two = FortyTwo()
+descriptors = {
+    row['MeSH Lowest Level Classification']: (
+        row['MeSH Classification Description'].replace("\n", "")
+        if row['MeSH Classification Description'] else ""
+    )
+    for row in forty_two.execute_query(descriptor_query)
+}
+
+predictions = pd.read_csv(os.path.join(here, f'../data/interim/{predictions_file}'))
+predicitons = predictions[predictions['Prob'] > 0.04]
+predictions.rename({'Grant id': 'Grant ID'}, axis=1, inplace=True)
+
+merged_predictions_metadata = pd.merge(left=grants, right=predictions, how='right', on='Grant ID')
+merged_predictions_metadata['MeSH Classification Description'] = \
+    merged_predictions_metadata['Tag'].apply(
+        lambda x: descriptors.get(x, "")
+)
+
+# Only End Date > 2016 (or active)
+merged_predictions_metadata = \
```

Comment:
this could be an argument

Reply:
I take the point of this being arguments, but I don't think I'll run this in any other file, other than the _exact_ file in `data/raw` or `data/interim` so I don't really want this name to change.